packages/core/src/modules/dif-presentation-exchange/DifPresentationExchangeService.ts

```diff
@@ -185,6 +183,42 @@ export class DifPresentationExchangeService {
         submission_requirements: undefined,
       }
 
+      if (presentationToCreate.claimFormat === ClaimFormat.MsoMdoc) {
+        if (presentationToCreate.verifiableCredentials.length !== 1) {
+          throw new DifPresentationExchangeError(
+            'Currently a Mdoc presentation can only be created from a single credential'
+          )
+        }
+        const mdocRecord = presentationToCreate.verifiableCredentials[0].credential
+        if (!openid4vp) {
+          throw new DifPresentationExchangeError('Missing openid4vp options for creating MDOC presentation.')
+        }
+
+        const { deviceResponseBase64Url, presentationSubmission } = await MdocDeviceResponse.openId4Vp(agentContext, {
+          mdocs: [Mdoc.fromBase64Url(mdocRecord.base64Url)],
+          presentationDefinition: presentationDefinition as DifPresentationExchangeDefinitionV2,
+          sessionTranscriptOptions: {
+            ...openid4vp,
+          },
+        })
+
+        verifiablePresentationResultsWithFormat.push({
+          verifiablePresentationResult: {
+            presentationSubmission: presentationSubmission,
+            verifiablePresentation: deviceResponseBase64Url,
+            presentationSubmissionLocation: PresentationSubmissionLocation.EXTERNAL,
+          },
+          claimFormat: presentationToCreate.claimFormat,
+        })
+
+        continue
```

Comment:
I think instead of `continue` we could use an else clause maybe? I had a hard time understanding this code, until i spotted this `continue` statement